master/pkg/schemas/expconf/storage_config.go

```diff
@@ -136,3 +138,17 @@ type S3ConfigV0 struct {
 type GCSConfigV0 struct {
 	RawBucket *string `json:"bucket"`
 }
+
+//go:generate ../gen.sh
+// AzureConfigV0 configures storing checkpoints on Azure.
+type AzureConfigV0 struct {
+	RawContainer        *string `json:"container"`
+	RawConnectionString *string `json:"connection_string"`
+	RawAccountURL       *string `json:"account_url"`
+	RawCredential       *string `json:"credential"`
+}
+
+// Merge implements schemas.Mergeable.
+func (a AzureConfigV0) Merge(other interface{}) interface{} {
+	return schemas.UnionMerge(a, other)
```

Comment:
This isn't a proper union type, because there are no `union` tags.  You will have to manually implement the correct merging behavior.